test/parallel/test-process-setuid-setgid.js

```diff
@@ -20,25 +20,49 @@
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 'use strict';
-// Requires special privileges
 const common = require('../common');
+
 const assert = require('assert');
 
-var oldgid = process.getgid();
-process.setgid('nobody');
-var newgid = process.getgid();
-assert.notStrictEqual(newgid, oldgid, 'gids expected to be different');
+if (common.isWindows) {
+  // uid/gid functions are POSIX only
+  assert.strictEqual(process.getuid, undefined);
+  assert.strictEqual(process.setuid, undefined);
+  assert.strictEqual(process.getgid, undefined);
+  assert.strictEqual(process.setgid, undefined);
+  return;
+}
 
-var olduid = process.getuid();
-process.setuid('nobody');
-var newuid = process.getuid();
-assert.notStrictEqual(newuid, olduid, 'uids expected to be different');
-
-try {
-  process.setuid('nobody1234');
-} catch (e) {
-  assert.strictEqual(e.message,
-               'failed to resolve group',
-               'unexpected error message'
+assert.throws(() => {
+  process.setuid('fhqwhgadshgnsdhjsdbkhsdabkfabkveybvf');
+}, /^Error: setuid user id does not exist$/);
+
+// If we're not running as super user...
+if (process.getuid() !== 0) {
+  assert.doesNotThrow(() => {
+    process.getgid();
+    process.getuid();
+  });
+
+  assert.throws(
+    () => { process.setgid('nobody'); },
+    /^Error: (EPERM, .+|setgid group id does not exist)$/
```

Comment:
> I am actually getting Error: EPERM, Operation not permitted for both the cases. 

@thefourtheye Error message depends on your operating system. SmartOS gives `Error: EPERM, Not owner` here.

The regexp allows any error that starts with `Error: EPERM, ` and also allows `Error: setgid group id does not exist` for the situation where `nobody` is not included on the operating system (which happens on ubuntu1404-64 in FIPS mode on our CI). 

TL;DR: Test passes if you have group `nobody` and get `EPERM` back along with some text from the OS, and it will also pass if group `nobody` doesn't exist at all.